csrc/python_frontend/translation.cpp

```diff
@@ -694,7 +702,40 @@ class FusionTranslator : public OptInConstDispatch {
             DataType>(rop),
         axes,
         /*keep_dim=*/false,
-        std::get<PrimDataType>(rop->out()->dtype().type)));
+        dtype));
+  }
+
+  // If input and output values share the same type, a LoadStoreOp will be
+  // created instead of a CastOp.
+  void handle(const LoadStoreOp* lsop) final {
+    // short-circuit: lsop is a permutation.
+    if (lsop->out()->isA<TensorView>() &&
+        lsop->out()->as<TensorView>()->hasRoot()) {
+      return handlePermute(lsop);
+    }
+
+    // Skip set unary operation
+    int64_t input_fid = map_val_to_fd_index_.at(lsop->in());
+    map_val_to_fd_index_.emplace(lsop->out(), input_fid);
+  }
+
+  void handlePermute(const LoadStoreOp* lsop) {
+    TensorView* out_tv = lsop->out()->as<TensorView>();
+    const std::vector<IterDomain*>& root_domain = out_tv->domain()->root();
+    std::vector<int64_t> new2old;
+    new2old.reserve(root_domain.size());
+    for (IterDomain* id : out_tv->domain()->logical()) {
+      auto root_id_iter = std::find(root_domain.begin(), root_domain.end(), id);
+      new2old.push_back(std::distance(root_domain.begin(), root_id_iter));
+    }
```

Comment:
Can you use `ir_utils::computePermutation` here?

Reply:
It works.